### ISSUE TITLE
[6.0.0-rc.0] TypeError: undefined is not iterable (in @angular/compiler-cli)

### ISSUE THREAD

**Issue by benneq** [angular/angular#22903]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
</code></pre>

## Current behavior

I get the following error:
```
[1] ERROR in : TypeError: undefined is not iterable
[1]     at mergeEmitResults (/Users/benneq/Development/ng6/node_modules/@angular/compiler-cli/src/transformers/program.js:844:22)
[1]     at AngularCompilerProgram._emitRender2 (/Users/benneq/Development/ng6/node_modules/@angular/compiler-cli/src/transformers/program.js:314:30)
[1]     at AngularCompilerProgram.emit (/Users/benneq/Development/ng6/node_modules/@angular/compiler-cli/src/transformers/program.js:201:18)
[1]     at AngularCompilerPlugin._emit (/Users/benneq/Development/ng6/node_modules/@ngtools/webpack/src/angular_compiler_plugin.js:781:49)
[1]     at Promise.resolve.then.then.then (/Users/benneq/Development/ng6/node_modules/@ngtools/webpack/src/angular_compiler_plugin.js:610:54)
[1]     at <anonymous>
[1]     at process._tickCallback (internal/process/next_tick.js:118:7)
```

I also did a `console.log` in `@angular/compiler-cli/src/transformers/program.js:844`. Here's the output:
```
[1] ℹ ｢wdm｣: Compiling...
[1]  10% building modules 0/1 modules 1 active ...t/ng6/src/app/app.module.ngfactory.js{ emitSkipped: false,
[1]   diagnostics: [],
[1]   emittedFiles: undefined,
[1]   sourceMaps: 
[1]    [ { sourceMapFilePath: '/Users/benneq/Development/ng6/src/app/artist/components/artist-create/artist-create.component.ngfactory.js.map',
[1]        jsSourceMappingURL: 'artist-create.component.ngfactory.js.map',
[1]        sourceMapFile: 'artist-create.component.ngfactory.js',
[1]        sourceMapSourceRoot: '',
[1]        sourceMapSources: [Array],
[1]        inputSourceFileNames: [Array],
[1]        sourceMapNames: [],
[1]        sourceMapMappings: ';;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;gECOgB,mMAA0D,IAAA,oDAAkB,iCAA5E,0DAA0D;gEAE9D,4EAAqB,KAAA,+IAErB;gEAYI,oMAA0D,IAAA,oDAAkB,iCAA5E,0DAA0D;gEAE9D,4EAAqB,KAAA,+IAErB;uEA3BZ;;;;;wBAAmD;MAAtB;;wBAAqB;MAAlD,ybAAmD,MAE3C,6KAAgB,IAAA,wDAAsB,MACtC,oLAAkB,IACd,40CAAgB,MACZ,+IAAW,IAAA,iCAAI,MACf;;;;;;;;;;;;;;;;;;;;;0kCAA8D,KAC9D,oLAA0D,IAE9D,uLAAqB,IAOjC,kMAAU,IACN;;;;;wBAAqF;MAA5D;;wBAAuB;MAAhD,8XAAqF,IACjF,kLAAgB,IAAA,wDAAsB,MACtC,yLAAkB,IACd,i1CAAgB,MACZ,gJAAW,IAAA,iCAAI,MACf;;;;;;;;;;;;;;;;;;;;;0kCAA8D,KAC9D,oLAA0D,IAE9D,uLAAqB,IAIzB,8NAAkB,IACd,8IAAyC,YAAvB;;wBAAsB;MAAxC,sKAAyC,MACzC,kIAAsF,EAAnE;;wBAA4B;MAA/C,0MAAsF,MAClF,2EAA0B,KAAA,gDAAmB,MAC7C,2EAA4B,KAAA,iEAAqB,iDAjCvD,mBAAkB,OAA5B,YAAmD,EAAzC,+BAAkB,OAA5B,YAAmD,EAAzC,SAAkB,GAMuC,mBAAQ,EAA3D,aAA8D,EAAX,UAAQ,GAA/B,uBAAsB,EAAlD,aAA8D,EAAlC,UAAsB,GAAC,mBAAQ,EAA3C,uBAAW,EAA3B,aAA8D,EAAX,UAAQ,EAA3C,UAAW,GAChB,wHAA8C,IAAzD,aAA0D,EAA/C,UAA8C,GAExD,oBAAe,UAApB,aAAqB,EAAhB,UAAe,GAQtB,oBAAkB,OAAxB,aAAqF,EAA/E,UAAkB,GAKuC,mBAAQ,EAA3D,aAA8D,EAAX,UAAQ,GAA/B,wBAAsB,EAAlD,aAA8D,EAAlC,WAAsB,GAAC,oBAAQ,EAA3C,wBAAW,EAA3B,aAA8D,EAAX,WAAQ,EAA3C,WAAW,GAChB,4HAA8C,IAAzD,aAA0D,EAA/C,WAA8C,GAExD,qBAAe,UAApB,aAAqB,EAAhB,WAAe,GAM4B,wCAAqC,WAArF,aAAsF,EAAtC,WAAqC,mDA/BjG,iaAAmD,GAIvC,86CAAgB,IAEZ,ygCAA8D,IAW1E,ubAAqF,GAG7E,g7CAAgB,IAEZ,ygCAA8D,IAOtE,+EAAkB,GAIkB,2GAAqB',
[1]        sourceMapSourcesContent: [Array],
[1]        sourceMapDecodedMappings: [Array] } ] }

```

## Expected behavior

No error.

## Minimal reproduction of the problem with instructions

I'm not entirely sure WHEN this error happens. I couldn't figure our a pattern yet.

**EDIT: Further investigations:**
I'm using `ng serve --aot` (with `es2015`).

The error occures when editing HTML and/or `ts` files. I'm not sure what triggers the error. Sometimes it happens when adding / removing comments from the HTML or adding / removing a line from my `.ts` Component.

A workaround is: add whitespace, remove whitespace, save file (to trigger recompilation). Now it compiles just fine without any error.

## What is the motivation / use case for changing the behavior?

It's a bug.

## Environment

<pre><code>
Angular version: 6.0.0-rc.0
Angular CLI: 6.0.0-beta.5 (because beta.6 has a bug with ng serve)
rxjs: 6.0.0-beta.0
rxjs-compat: 6.0.0-beta.1
Typescript: 2.7.2

Browser:
- [x] Chrome (desktop) version 65
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [ ] Firefox version XX
- [ ] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX
 
For Tooling issues:
- Node version: v9.8.0 (and v9.7.1 tested)  
- Platform:  macOS 10.13.3 

Others:

</code></pre>

**Comment by benneq** [comment]
I now set up a completely new project using:
```
Angular version: 6.0.0-rc.0
Angular CLI: 6.0.0-beta.6
Angular Material: 6.0.0-beta.5
Angular Flex Layout: 5.0.0-beta.13
rxjs: 6.0.0-beta.1
rxjs-compat: 6.0.0-beta.1
Typescript: 2.7.2
```

Then copied my whole old project into the new project. Now everything is working.

The only difference is, that I'm now using `es5` instead of `es2015`, **no** `aot` (because of a Bug withing Angular CLI beta.6), and a newer version of Angular CLI and Angular Material.

You can close this issue if you want. Because I don't know if this is a "real" bug or was just an unlucky combination of dependencies and settings.

**Comment by floreks** [comment]
Unfortunately, we are encountering the same issue after bumping angular from `5.2.X` to `6.0.0-rc.0`. There is no real pattern, that I can see. I can simply add new line, hit save to trigger recompilation and there is like 50% chance I will get this error. After removing empty line it might happen again or not.

I'm not sure if it will help but here are our dependencies: https://github.com/kubernetes/dashboard/blob/angular-migration/package.json

**Comment by patrickmichalina** [comment]
Having this issue as well preventing AOT us in a large application. Does NOT appear to happen in ultra basic hello world apps. IE11 specific for me.

**Comment by JoostK** [comment]
This has been fixed in #22899, coincidentally the same day this was reported.
